vignettes/understanding_shapr_vaeac.Rmd

```diff
@@ -0,0 +1,981 @@
+  vaeac.num_vaeacs_initiate = 2,
+  vaeac.extra_parameters = list(
+    vaeac.save_every_nth_epoch = vaeac.save_every_nth_epoch,
+    vaeac.verbose = TRUE
+  )
+)
+all.equal(expl_without_messages$shapley_values, expl_with_messages$shapley_values)
+```
+
+
+
+Let us take a look at when we using `progressr`.
+```{r progressr true verbose false, cache = TRUE}
+progressr::handlers("cli")
+# If no progression handler is specified, the txtprogressbar is used
+# Other progression handlers:
+# progressr::handlers('rstudio') # requires the 'rstudioapi' package
+# progressr::handlers('handler_winprogressbar') # Window only
+# progressr::handlers('cli') # requires the 'cli' package
+#
+# Another progressbar with sound
+# handlers(list(
+#   handler_pbcol(
+#       adjust = 1.0,
+#     complete = function(s) cli::bg_red(cli::col_black(s)),
+#   incomplete = function(s) cli::bg_cyan(cli::col_black(s))
+# ),
+#   handler_beepr(
+#     finish   = "wilhelm",
+#     interval = 2.0
+#   )
+# ))
+with_progress({
+  expl_with_progressr <- explain(
+    model = model,
+    x_explain = x_explain,
+    x_train = x_train,
+    approach = approach,
+    prediction_zero = p0,
+    n_combinations = n_combinations,
+    n_samples = 100,
+    n_batches = 5,
+    seed = seed,
+    vaeac.epochs = 10,
+    vaeac.num_vaeacs_initiate = 2,
+    vaeac.extra_parameters = list(
+      vaeac.save_every_nth_epoch = vaeac.save_every_nth_epoch,
+      vaeac.verbose = FALSE
+    )
+  )
+})
+all.equal(expl_without_messages$shapley_values, expl_with_progressr$shapley_values)
+```
+
+We can also use `progressr` together with the `progress` bars that are created 
+when `vaeac.verbose == TRUE`, but then we will only see the `progressr` get updates
+while the `progress` bars will just be printed. Note that there an be some minor visual 
+artifacts.
+```{r progressr true verbose true, cache = TRUE}
+handlers("progress")
+with_progress({
+  expl_with_progressr_verbose <- explain(
+    model = model,
+    x_explain = x_explain,
+    x_train = x_train,
+    approach = approach,
+    prediction_zero = p0,
+    n_combinations = n_combinations,
+    n_samples = 100,
+    n_batches = 5,
+    seed = seed,
+    vaeac.epochs = 10,
+    vaeac.num_vaeacs_initiate = 2,
+    vaeac.extra_parameters = list(
+      vaeac.save_every_nth_epoch = vaeac.save_every_nth_epoch,
+      vaeac.verbose = TRUE
+    )
+  )
+})
+all.equal(expl_without_messages$shapley_values, expl_with_progressr_verbose$shapley_values)
+```
+
+
+## Continue the training of the vaeac approach
+
+In the case the user has set a too low number of training epochs and sees that the network is still learning,
+then the user can continue to train the network from where it stopped. Thus, a good workflow can therefore
+be to call the `explain()` function with a `n_samples = 1` (to not waste to much time to generate MC samples), 
+then look at the training and evaluation plots of the `vaeac`. If not satisfied, then train more. If satisfied,
```

Comment:
Hmmm. Then explain will have a specific if clause just for vaeac. Would it be better if this was a global explain parameter? It would be nice to return the ctree models too. What do you think? I'll just keep it as it is. This is also one of the main reasons for why I wanted the train_vaeac function to be a separate function that is not too much connected to explain()/shapr.